### PR TITLE
test(RHINENG-20813): User able to bulk delete empty workspaces

### PR DESCRIPTION
**Pull request by adarshdubey-star**
Jira - https://issues.redhat.com/browse/RHINENG-20813

## Summary by Sourcery

Tests:
- Add a Playwright test that creates multiple empty workspaces, performs a bulk delete action, and verifies they are removed

### CODE REVIEW

File: _playwright-tests/test_workspace.test.ts [hunk 1]
```diff
@@ -125,3 +125,65 @@ test('User cannot delete a workspace with systems', async ({ page }) => {
...
+    await expect(bulkSelectCheckbox).toBeChecked();
+
+    // Click the Bulk Actions toggle button
+    const bulkActionsToggle = page.locator('[data-ouia-component-id="BulkActionsToggle"]');
+    await expect(bulkActionsToggle).toBeVisible();
+    await bulkActionsToggle.click();
+
+    await page.getByRole('menuitem', { name: 'Delete workspaces' }).first().click();
+    await expect(dialog).toBeVisible();
+    await dialog.locator('input[type="checkbox"]').first().check();
```
Comment (computercamplove) [inline _playwright-tests/test_workspace.test.ts:183]: no need this line - the checkbox is removed here https://github.com/RedHatInsights/insights-inventory-frontend/commit/8ffa46a3540e9f89d81df70b1c0d74a02d70cc45

File: _playwright-tests/test_workspace.test.ts [hunk 2]
```diff
@@ -125,3 +125,65 @@ test('User cannot delete a workspace with systems', async ({ page }) => {
...
+     */
+    
+    // Close any modals/popups before navigating
+    await closePopupsIfExist(page);
+
+    // Navigate to the Workspaces page
+    await navigateToWorkspacesFunc(page);
+    const dialog = page.locator('[role="dialog"]');
+
+    // Create 5 workspaces
```
[the remark this answers is not in the document]
Comment (computercamplove) [reply _playwright-tests/test_workspace.test.ts:148]: I think this could even be its own Jira task. Setting up tests properly is an important part of ensuring they are stable and fast, so it's something we'll need to look into anyway.
You can go ahead and resolve this for now - I'll create the Jira ticket tomorrow.

File: _playwright-tests/test_navigation.test.ts [hunk 3]
```diff
@@ -20,15 +20,13 @@ test('User can navigate to the Staleness and Deletion page via the menu', async
...
   await stalenessLink.click();
-
   await expect(page.getByRole('heading', { name: 'Staleness and Deletion' })).toBeVisible({ timeout: 90000 });
 
 });
 
-
 test('User can navigate to the workspaces page via the menu', async ({ page }) => {
   // Optionally close any modals/popups before navigating
   await closePopupsIfExist(page);
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply _playwright-tests/test_navigation.test.ts:32]: Docstrings within the functions themselves, you mean

**Comment by adarshdubey-star** [comment]
Hey @computercamplove @raswonders, I have updated the whole file with the test.step() let's keep it intact.

File: _playwright-tests/test_navigation.test.ts [hunk 4]
```diff
@@ -1,45 +1,52 @@
...
-
 import { navigateToInventoryFunc } from './helpers/navHelpers';
 import { closePopupsIfExist } from './helpers/loginHelpers';
 
 test('User can navigate to the inventory page', async ({ page }) => {
-  // Optionally close any modals/popups before navigating
-  await closePopupsIfExist(page);
-  // Navigate directly to the inventory page
-  await navigateToInventoryFunc(page);
+  await test.step('Close popups and navigate to Inventory page', async () => {
```
Comment (computercamplove) [inline _playwright-tests/test_navigation.test.ts:6]: since test has 1 step, it is not necessary here, but i am leaving it on you - if you want to leave it or not

File: _playwright-tests/test_navigation.test.ts [hunk 5]
```diff
@@ -1,45 +1,52 @@
...
-
+  await test.step('Close popups and navigate to Inventory page', async () => {
+    await closePopupsIfExist(page);
+    await navigateToInventoryFunc(page);
+  });
+
+  await test.step('Click on Workspaces menu item and reload page', async () => {
+    const WorkspacesParam = page.locator('[data-ouia-component-id="Workspaces"]');
+    await WorkspacesParam.click();
+    await page.reload({ waitUntil: 'networkidle' });
```
Comment (computercamplove) [inline _playwright-tests/test_navigation.test.ts:44]: thank you for adding this here - i think we need to report this issue to the devs - why we have this issue with Workplaces page that we require reload the page

File: _playwright-tests/test_workspace.test.ts [hunk 6]
```diff
@@ -1,126 +1,189 @@
...
+    await page.getByRole('menuitem', { name: 'Rename workspace' }).first().click();
+
+    const dialog = page.locator('[role="dialog"]');
+    await expect(dialog).toBeVisible();
+    await dialog.locator('input').first().fill(RenamedWorkspace);
+    await dialog.getByRole('button', { name: 'Save' }).click();
+  });
+
+  await test.step('Verify rename was successful', async () => {
+    await navigateToWorkspacesFunc(page);
```
Comment (computercamplove) [inline _playwright-tests/test_workspace.test.ts:61]: since you are testing rename and delete from Workspace details - you can verify renaming alredy inside of workspace via checking the title:

`await expect(page.getByRole('heading', { name: renamedWorkspace })).toBeVisible();`

after that you wouldn't need extra step in _"Delete renamed workspace"_ to navigate back to details, you can remove this block of code:
```
const renamedLink = page.getByRole('link', { name: RenamedWorkspace });
await expect(renamedLink).toBeVisible({ timeout: 100000 });
await renamedLink.click();
```

File: _playwright-tests/test_workspace.test.ts [hunk 7]
```diff
@@ -1,126 +1,189 @@
...
+    await expect(searchInput).toBeVisible();
+    await page.reload({ waitUntil: 'networkidle' });
+    await searchInput.fill(WorkspaceName);
+
+    const workspaceLink = page.getByRole('link', { name: WorkspaceName });
+    await expect(workspaceLink).toBeVisible({ timeout: 100000 });
+    await workspaceLink.click();
+  });
+
+  await test.step('Attempt to delete workspace with systems and verify error', async () => {
```
Comment (computercamplove) [inline _playwright-tests/test_workspace.test.ts:122]: maybe `warning` instead of `error` , since it intended this way

File: _playwright-tests/test_navigation.test.ts (hunk 5 above)
Comment (adarshdubey-star) [reply _playwright-tests/test_navigation.test.ts:44]: Yeah somehow when you switch between workspaces to systems from the menu ,it comes back to systems page

File: _playwright-tests/test_workspace.test.ts (hunk 6 above)
Comment (adarshdubey-star) [reply _playwright-tests/test_workspace.test.ts:60]: Okay let me try something

**Comment by adarshdubey-star** [comment]
@computercamplove, I have removed the `await closePopupsIfExist(page);` from the files I have requested a change in your PR to make [this](https://github.com/RedHatInsights/insights-inventory-frontend/pull/2660/files#diff-c9e5700c17cdfc3ef13cc4b889a6f087b6fba760e2f696a211eedf72d96c85caR8) global

File: _playwright-tests/test_workspace.test.ts (hunk 6 above)
Comment (adarshdubey-star) [reply _playwright-tests/test_workspace.test.ts]: updated

**Comment by computercamplove** [comment]
> @computercamplove, I have removed the `await closePopupsIfExist(page);` from the files I have requested a change in your PR to make [this](https://github.com/RedHatInsights/insights-inventory-frontend/pull/2660/files#diff-c9e5700c17cdfc3ef13cc4b889a6f087b6fba760e2f696a211eedf72d96c85caR8) global

my PR is not ready, it will not be issue without it for while? you can already make this change in your PR - i am fine with it!
